File: Vanilla/sources/zdoc/media/lua/shared/Library/SteamFriend.lua
---@class SteamFriend : zombie.core.znet.SteamFriend
---@field private name String
---@field private steamID long
---@field private steamIDString String
SteamFriend = {}

---@public
---@return Texture
function SteamFriend:getAvatar() end

---@public
---@return String
function SteamFriend:getState() end

---@public
---@return String
function SteamFriend:getName() end

---@public
---@return String
function SteamFriend:getSteamID() end

File: Vanilla/sources/zomboid/zombie/core/znet/SteamFriend.java
package zombie.core.znet;

import zombie.core.textures.Texture;

public class SteamFriend {
   private String name = "";
   private long steamID;
   private String steamIDString;

   public SteamFriend() {
   }

   public SteamFriend(String var1, long var2) {
      this.steamID = var2;
      this.steamIDString = SteamUtils.convertSteamIDToString(var2);
      this.name = var1;
   }

   public String getName() {
      return this.name;
   }

   public String getSteamID() {
      return this.steamIDString;
   }

   public Texture getAvatar() {
      return Texture.getSteamAvatar(this.steamID);
   }

   public String getState() {
      int var1 = SteamFriends.GetFriendPersonaState(this.steamID);
      switch(var1) {
      case 0:
         return "Offline";
      case 1:
         return "Online";
      case 2:
         return "Busy";
      case 3:
         return "Away";
      case 4:
         return "Snooze";
      case 5:
         return "LookingToTrade";
      case 6:
         return "LookingToPlay";
      default:
         return "Unknown";
      }
   }
}
